arbitrum-docs/devs-how-tos/how-to-estimate-gas.mdx

```diff
@@ -0,0 +1,142 @@
+    - Call `NodeInterface.GasEstimateComponents()` with the transaction data and subtract the second element (`gasEstimateForL1`, which estimates the L1 part of the fees) from the first (`gasEstimate`, which includes both the L1 and the L2 parts).
+- L1P (L1 estimated price per byte of data) ⇒ Estimated cost of posting 1 byte of data on L1:
+    - Call `ArbGasInfo.getPricesInWei()` and get the second element
+- L1S (Size of data to be posted on L1, in bytes) ⇒ This will depend on the data of the transaction. Keep in mind that Arbitrum adds a fixed amount to this number to make up for the static part of the transaction, which is also posted on L1 (140 bytes).
+    - For Arbitrum Nova (AnyTrust), the size of the data is also a fixed value, as only the Data Availability Certificate is posted on L1, [as explained here](/inside-anytrust#data-availability-certificates).
+
+## An example of how to apply this formula in your code
+
+Finally, we show an example of how to get the values we just described and how to estimate the gas usage of a transaction in Javascript. We will use our [SDK](https://github.com/OffchainLabs/arbitrum-sdk) to connect to `ArbGasInfo` and `NodeInterface`.
+
+We first instantiate a factory object for ArbGasInfo and NodeInterface, using two methods from the SDK. `l2Provider` is a regular JSON RPC provider for the L2 network we are using, and both `ARB_GAS_INFO` and `NODE_INTERFACE_ADDRESS` are the addresses of the precompiles in said network.
+
+```ts
+const { ArbGasInfo__factory } = require("@arbitrum/sdk/dist/lib/abi/factories/ArbGasInfo__factory");
+const { NodeInterface__factory } = require("@arbitrum/sdk/dist/lib/abi/factories/NodeInterface__factory");
+const { ARB_GAS_INFO, NODE_INTERFACE_ADDRESS } = require("@arbitrum/sdk/dist/lib/dataEntities/constants");
+
+...
+
+// Instantiation the ArbGasInfo and NodeInterface objects
+const arbGasInfo = ArbGasInfo__factory.connect(
+    ARB_GAS_INFO,
+    l2Provider
+);
+const nodeInterface = NodeInterface__factory.connect(
+    NODE_INTERFACE_ADDRESS,
+    l2Provider
+);
+```
+
+For this example, we will use the method `ArbGasInfo.getPricesInWei()` and `NodeInterface.gasEstimateComponents()` to get all the information we need. For the gasEstimateComponents() call we will pass a destinationAddress (might be any address) and the data we want to send, to get results as accurate as possible.
+
+```ts
+// Getting the gas prices from ArbGasInfo.getPricesInWei()
+const gasComponents = await arbGasInfo.callStatic.getPricesInWei();
+
+// And the estimations from NodeInterface.GasEstimateComponents()
+const gasEstimateComponents = await nodeInterface.callStatic.gasEstimateComponents(
+    destinationAddress,
+    false,
+    txData
+);
+```
+
+With this, we can now get the values of the 4 variables we will use in our formula.
```

Comment:
```suggestion
With this, we can now get the values of the 4 variables we'll use in our formula:
```

I'll do periodic scans for this type of editorial consistency, eg:

 - using conversational language / contractions
 - using colons to explicitly connect "this thing" to "the next thing"